### PR TITLE
feat: support http response streaming

### PR DESCRIPTION
**Pull request by jakoblorz**
Needed to get Server-Sent Events running using remix-electron, after some debugging and patching I got this implemented publicly here. Now also patterns like [remix-utils/server-sent-events](https://github.com/sergiodxa/remix-utils#server-sent-events) are working. Might also enable faster rendering with streaming responses now (not tested though and not the focus of my implementation).

Took some time to update the lock file and ~make sure~ tried that I don't break any test cases. Needed to make some adjustments down the line regarding tooling as `pnpm` does not accept the current state of the lockfile. The only thing that I don't get to run are the linux build test cases, I suspect, again, some issue with tooling (see [here](https://github.com/jakoblorz/remix-electron/actions/runs/5032744316/jobs/9026605974))

### CODE REVIEW

File: src/asset-files.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ export async function serveAsset(
   if (!file) return
 
   return {
-    data: await file.content(),
+    data: Readable.from(await file.content()),
```
Comment (jakoblorz) [inline src/asset-files.ts:35]: Not sure what to do here, AssetFile does not feature the typical `Blob`-style stream API, so we still need to read the contents first and then create a `Readable` from that.

File: src/serve-remix-response.ts [hunk 2]
```diff
@@ -27,8 +36,17 @@ export async function serveRemixResponse(
     values.push(value)
   }
 
+  if (response.body instanceof ReadableStream) {
+    return {
+      // @ts-expect-error
+      data: Readable.from(response.body),
```
Comment (jakoblorz) [inline src/serve-remix-response.ts:42]: Don't like this here, but it works. `Readable.from` expects an `Iterable` and response.body is already a `ReadableStream<UInt8Array>`, but only in this constellation it works.

File: src/asset-files.ts (hunk 1 above)
Comment (jakoblorz) [reply src/asset-files.ts]: Uses `fs.createReadStream` now

**Comment by itsMapleLeaf** [comment]
Thanks for this! I'll take a closer look at this later. The Linux error might be a fluke

**Comment by itsMapleLeaf** [comment]
It looks like some of the workflow dependencies were out of date, so I updated them on main and it seems to be good now. @jakoblorz After the merge conflict is fixed, I think this'll be good to merge 👍

**Comment by jakoblorz** [comment]
Alright, nice! Do you prefer to have the added dependency (`@remix-run/web-stream`) as direct or as peer dependency? Having it as a direct dependency makes it easier to upgrade, but I would prefer it as a peer tbh.

**Comment by jakoblorz** [comment]
Hell yeah, in my own CI it is now running through without issues :fire:
https://github.com/jakoblorz/remix-electron/actions/runs/5039783313/jobs/9038145258

**Comment by itsMapleLeaf** [comment]
> And good work updating all these dependencies btw @itsMapleLeaf, I saw the history 😄

Thanks haha

> Do you prefer to have the added dependency (@remix-run/web-stream) as direct or as peer dependency? 

Since its inclusion seems to be a Node compatibility thing, I think an optional peer makes sense

**Pull request merged**